src/Compilers/Core/CodeAnalysisTest/Diagnostics/OperationTests.cs

```diff
@@ -272,5 +272,90 @@ private void TestGetFlowGraphInvalidArgumentCore(string argumentExceptionMessage
                 Assert.Equal(new ArgumentException(argumentExceptionMessage, "methodBody").Message, ex.Message);
             }
         }
+
+        [CompilerTrait(CompilerFeature.IOperation, CompilerFeature.Dataflow)]
+        [Fact]
+        public void TestControlFlowGraphCreateFromSyntax()
+        {
+            var source = @"
+class C
+{
+    void M(int x)
+    {
+        x = 0;
+    }
+}";
+            var tree = CSharpSyntaxTree.ParseText(source);
+            var options = tree.Options.WithFeatures(new[] { new KeyValuePair<string, string>("flow-analysis", "true") });
+            tree = tree.WithRootAndOptions(tree.GetRoot(), options);
+            var compilation = CSharpCompilation.Create("c", new[] { tree });
+            var model = compilation.GetSemanticModel(tree, ignoreAccessibility: true);
+            var methodBodySyntax = tree.GetCompilationUnitRoot().DescendantNodes().OfType<BaseMethodDeclarationSyntax>().Last();
+
+            // Verify ArgumentNullException
+            try
+            {
+                _ = ControlFlowGraph.Create(node: null, model);
+            }
+            catch (ArgumentNullException ex)
+            {
+                Assert.Equal(ex.Message, new ArgumentNullException("node").Message);
+            }
+
+            try
+            {
+                _ = ControlFlowGraph.Create(methodBodySyntax, semanticModel: null);
+            }
+            catch (ArgumentNullException ex)
+            {
+                Assert.Equal(ex.Message, new ArgumentNullException("semanticModel").Message);
+            }
+
+            // Verify argument exception on providing a syntax node in executable code which does not produce root operation.
```

Comment:
This doesn't actually verify that the exception was thrown. Please use `Assert.Throws`

Reply:
> Please use Assert.Throws

Alternatively a boolean variable can be used to assert that the catch was hit. ```Assert.Throws``` probably will not allow us to verify content of the exception.

---
In reply to: [198233158](https://github.com/dotnet/roslyn/pull/28079#discussion_r198233158) [](ancestors = 198233158)

Reply:
True, it does allow you to specify a variable name for ArgumentExceptions, but doesn't allow you to assert the whole message. Please add either a boolean flag, or add an Assert.Fail after the call to Create

---
In reply to: [198237384](https://github.com/dotnet/roslyn/pull/28079#discussion_r198237384) [](ancestors = 198237384,198233158)

Reply:
Yes, planning to add a bool check. I will also update the test above it which needs similar changes.